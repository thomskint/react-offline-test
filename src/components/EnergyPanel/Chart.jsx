import React from "react";
import PropTypes from "prop-types";
import ChartistGraph from "react-chartist";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
const useStyles = makeStyles(theme => ({
  graphStyle: {
    "& .ct-label": {
      fill: "black",
      fontWeight: "bold",
      fontSize: "1em",
      textTransform: "capitalize"
    }
  }
}));

const Chart = ({ generationmix }) => {
  const classes = useStyles();
  if (!generationmix)
    return <Typography variant="subtitle1">Error Creating Graph</Typography>;

  const data = {
    labels: generationmix.map(item => item.fuel),
    series: generationmix.map(item => item.perc)
  };

  const responsiveOptions = [
    [
      "screen and (min-width: 640px)",
      {
        chartPadding: 30,
        height: 500,
        labelOffset: 100,
        labelDirection: "explode",
        labelInterpolationFnc: function(value, idx) {
          return `${value} ${generationmix[idx].perc} %`;
        }
      }
    ],
    [
      "screen and (min-width: 1024px)",
      {
        labelOffset: 100,
        chartPadding: 20
      }
    ]
  ];

  const type = "Pie";
  return (
    <ChartistGraph
      className={classes.graphStyle}
      data={data}
      type={type}
      responsiveOptions={responsiveOptions}
    />
  );
};

PropTypes.Chart = {
  generationmix: PropTypes.array
};

export default Chart;
