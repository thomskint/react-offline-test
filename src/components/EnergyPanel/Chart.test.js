import React from "react";
import { shallow, mount } from "enzyme";
import jsonData from "./../../../example_api_response.json";
import Chart from "./Chart";
import ChartistGraph from "react-chartist";
describe("<Chart/> Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Chart />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("should not render chartist if no data passed and show error", () => {
    const wrapper = mount(<Chart />);
    expect(wrapper.html()).toEqual(
      '<h6 class="MuiTypography-root MuiTypography-subtitle1">Error Creating Graph</h6>'
    );
    wrapper.unmount();
  });
  test("should render chartist graph", () => {
    const {
      data: { generationmix }
    } = jsonData;
    const wrapper = mount(<Chart generationmix={generationmix} />);
    expect(wrapper.find(ChartistGraph)).toBeTruthy();
    wrapper.unmount();
  });
});
