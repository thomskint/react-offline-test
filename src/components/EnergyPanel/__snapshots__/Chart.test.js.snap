// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart/> Component renders correctly 1`] = `ShallowWrapper {}`;
